api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -22,140 +22,232 @@
+
+        Assert.assertEquals(formatter.format(new Date()) + "-Unk" + "-001", visitNumber);
+
+    }
+
+    @Test
+    public void assignVisitNumber_shouldAssignPatientQueueNewVisitNumberWhenNoPatientQueueExistingOnSameDate() {
+        PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+
+        Patient patient = Context.getPatientService().getPatient(10000);
+
+        Location location = Context.getLocationService().getLocation(1);
+
+        PatientQueue patientQueue = new PatientQueue();
+        patientQueue.setPatient(patient);
+        patientQueue.setStatus(PatientQueue.Status.PENDING);
+        patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue.setLocationFrom(location);
+        patientQueue.setLocationTo(location);
+        patientQueue.setPriority(0);
+        patientQueue.setPriorityComment("Emergency");
+        patientQueueingService.assignVisitNumber(patientQueue);
+
+        String newQueueNumber = patientQueueingService.generateVisitNumber(location, patient);
+
+        String visitNumber = patientQueueingService.generateVisitNumber(location, patient);
+
+        Assert.assertEquals(newQueueNumber, visitNumber);
+
+    }
+
+    @Test
+    public void assignVisitNumber_shouldAssignPatientQueueExistingVisitNumberWhenPatientQueueExistsOnSameDate() throws ParseException {
+        PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+
+        Patient patient = Context.getPatientService().getPatient(10000);
+
+        Location location = Context.getLocationService().getLocation(1);
+
+        PatientQueue patientQueue = new PatientQueue();
+        patientQueue.setPatient(patient);
+        patientQueue.setStatus(PatientQueue.Status.PENDING);
+        patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue.setLocationFrom(location);
+        patientQueue.setLocationTo(location);
+        patientQueue.setPriority(0);
+        patientQueue.setPriorityComment("Emergency");
+        patientQueueingService.assignVisitNumber(patientQueue);
+
+
+        PatientQueue patientQueue2 = new PatientQueue();
+        patientQueue2.setPatient(patient);
+        patientQueue2.setStatus(PatientQueue.Status.PENDING);
+        patientQueue2.setEncounter(Context.getEncounterService().getEncounter(10000));
+        patientQueue2.setLocationFrom(location);
+        patientQueue2.setLocationTo(location);
+        patientQueue2.setPriority(0);
+        patientQueue2.setPriorityComment("Emergency");
+        patientQueueingService.assignVisitNumber(patientQueue2);
+        patientQueueingService.savePatientQue(patientQueue2);
+
+        Assert.assertEquals(patientQueue.getVisitNumber(), patientQueue2.getVisitNumber());
+    }
+
+    @Test
+    public void assignVisitNumber_shouldReturnLastPatientQueueCreated() throws ParseException {
```

Comment:
Is assignVisitNumber's responsibility to assign a visit number or return last patient queue?